flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -1285,6 +1287,13 @@ public void setName(String name) {
 		customName = name;
 	}
 
+	public void setApplicationType(String type) {
+		if (type == null) {
```

Comment:
Let's use `org.apache.flink.util.Preconditions`:

`this.applicationType = Preconditions.checkNotNull(applicationType, "applicationType must not be null");`

`setName` should also be refactored in a separate commit.